app/src/main/java/org/thoughtcrime/securesms/conversation/v2/messages/VisibleMessageContentView.kt

```diff
@@ -310,6 +313,18 @@ class VisibleMessageContentView : ConstraintLayout {
         fun getTextColor(context: Context, message: MessageRecord): Int = context.getColorFromAttr(
             if (message.isOutgoing) R.attr.message_sent_text_color else R.attr.message_received_text_color
         )
+
+        // Method to grab the appropriate attribute for a message colour.
+        // Note: This is an attribute, NOT a resource Id - see `getColorResourceIdFromAttr` for that.
+        fun getMessageTextColourAttr(messageIsOutgoing: Boolean): Int =
+            if (messageIsOutgoing) R.attr.message_sent_text_color else R.attr.message_received_text_color
+
+        // Method to get an actual R.id.<SOME_COLOUR> resource Id from an attribute such as R.attr.message_sent_text_color etc.
+        fun getColorResourceIdFromAttr(context: Context, attr: Int): Int {
+            val typedValue = TypedValue()
+            context.theme.resolveAttribute(attr, typedValue, true)
+            return typedValue.resourceId
```

Comment:
if you like...

```
fun getColorResourceIdFromAttr(context: Context, attr: Int) = TypedValue().also {.context.theme.resolveAttribute(attr, it, true) }.resourceId
```

if this is an instance method you should already have `context`, otherwise, if this is a reusable utility it can go in some util package, and it could be an extension function on `Context`.

```
fun Context.getColorResourceIdFromAttr(attr: Int) = TypedValue().also { theme.resolveAttribute(attr, it, true) }.resourceId
```

Reply:
and same for above method